flink-core/src/main/java/org/apache/flink/api/common/state/StateDescriptor.java

```diff
@@ -249,12 +257,13 @@ public boolean isSerializerInitialized() {
 	 */
 	public void initializeSerializerUnlessSet(ExecutionConfig executionConfig) {
```

Comment:
This is slightly orthogonal to this change, but: could we get rid of this method and instead change `getSerializer()` to `getSerializer(ExecutionConfig)`. That way, we don't have to be concerned about forgetting to call `initializeSerializerUnlessSet()`.